test.js

```diff
@@ -0,0 +1,62 @@
+var tests = {
+    	/*
+	Given an integer array of size N, compute the sum of all even numbers in this array
+	*/
+	firstTest: function () {
+    		var array = [1, 10, 2, 9, 3, 8, 4, 7, 5, 6];
+	    	var result = 0;
+	    
+		for (var i = 0; i < array.length; i++) {
+			if (array[i]%2 == 0) {
+	  			result += array[i];
+		  	}
+		}
+	    	return result;
+	},
+
+    	/*
+	Given an integer array of size N, identify the largest 2 numbers in this array. Do not sort the array.
+	*/
+    	secondTest: function () { 
+    		var array = [1, 10, 2, 9, 3, 8, 4, 7, 5, 6, 11, 4];
+    		var biggestNr = 0;
+    		var secondNr = 0;
+
+    		for (var i = 0; i < array.length; i++) {
+    			if(array[i] > biggestNr) {
+    				secondNr = biggestNr;
+    				biggestNr = array[i];
+    			} 
+    		}
+    		return biggestNr + ' & ' + secondNr;
+    	},
+
+    	/*
+	Given an array of numbers having size N in which every number is between 1 and N, determine 
+	if there are any duplicates in it. You are allowed to destroy the array if you like. 
+	Do not sort the array or use bit vectors. Try to work within the array without using any other 
+	temporary data structures. Do not use 2 nesting for loops since the time complexity would be high. 
+	Try to find a solution by executing just one loop. Hint: manipulate the elements of the same array
+	as you loop through.
+	*/
+    	thirdTest: function () {
+    		var array = [1, 10, 2, 9, 5, 4, 4, 7, 5, 6, 3, 10, 8];
+		var result = '';
+
+		array.reduce( function(numbers, element) {
+			if (numbers.indexOf(element) < 0) {
+				numbers.push(element);
```

Comment:
`Try to work within the array without using any other temporary data structures`

`numbers` is a temporary data structure, there are solution without it.